src/node/internal/internal_errors.ts

```diff
@@ -89,6 +90,19 @@ export class NodeTypeError extends NodeErrorAbstraction implements TypeError {
   }
 }
 
+export class NodeSyntaxError
+  extends NodeErrorAbstraction
+  implements SyntaxError
+{
+  constructor(code: string, message: string) {
+    super(SyntaxError.prototype.name, code, message);
+    Object.setPrototypeOf(this, SyntaxError.prototype);
+    this.toString = function () {
+      return `${this.name} [${this.code}]: ${this.message}`;
+    };
```

Comment:
Why does toString need to be injected like this, instead of being declared as a method?

Reply:
No specific way. We can do the other way around as well.